draft-ietf-dnsop-svcb-https.md

```diff
@@ -770,8 +770,27 @@ ALPN IDs that do not uniquely identify a protocol suite (e.g. an ID that
 can be used with both TLS and DTLS) are not compatible with this
 SvcParamKey and MUST NOT be included in the ALPN set.
 
+To establish a connection to the endpoint, clients MUST
+
+1. Identify all supported IDs in the ALPN set.
+2. Collect the set of transports (e.g. TLS, DTLS, QUIC) implied by those IDs.
+3. For each transport in this set, construct a ProtocolNameList containing all
+the client's supported IDs for that transport, without regard to the ALPN set
+from SVCB.
+
+For example, if the ALPN set is \["http/1.1", "h3"\], the client could attempt
+to connect using TLS over TCP with a ProtocolNameList of \["http/1.1", "h2"\],
+and could also attempt a connection using QUIC, with a ProtocolNameList of
+\["h3"\].
+
+This procedure prevents downgrade attacks within a transport and ensures that
```

Comment:
FWIW, I find "downgrade attacks within a transport" somewhat confusing. Could we add an example?

Reply:
OK, I replaced that language with a more explicit definition.

Reply:
Much clearer -- thanks!